File: 283-MoveZeroes/283-MoveZeroes.java
// Last updated: 1/3/2026, 6:26:09 PM
class Solution {
    public void moveZeroes(int[] nums) {
        int j = 0;  // Pointer for the position of the next non-zero element

        // Move non-zero elements to the front
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != 0) {
                nums[j] = nums[i];  // Place non-zero element at position j
                if (i != j) {
                    nums[i] = 0;  // Set the current position i to 0
                }
                j++;  // Increment the position for the next non-zero element
            }
        }
    }
}

File: 283-MoveZeroes/283-MoveZeroes.cpp
// Last updated: 1/3/2026, 6:26:06 PM
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int i = 0, j = 0;
        int n = nums.size();
        while(j < n){
            if(nums[j] != 0){
                swap(nums[j], nums[i]);
                i++;
            }
            j++;
        }
    }
};